internal/crd/cache.go

```diff
@@ -33,4 +44,13 @@ func (c *Cache) Get(key string) (apiextensionsv1.CustomResourceDefinition, bool)
 
 func (c *Cache) Add(key string, value apiextensionsv1.CustomResourceDefinition) {
 	c.cache.Store(key, value)
+	if c.metrics == nil {
+		return
+	}
+	length := 0
+	c.cache.Range(func(key, value interface{}) bool {
+		length++
+		return true
+	})
+	c.metrics.UpdateCrdTotal(length)
 }
```

Comment:
Was just curious and took a quick glimpse already. It may be possible to replace `cache.Store` with [`cache.Swap`](https://pkg.go.dev/sync#Map.Swap). Then the `length` may be a property of the `Cache` struct that is increased whenever the `loaded` return value returns `false` => we get rid of iterating over the whole cache each time we add a value.

`length` would need to be a thread-safe counter though

Reply:
That is an excellent idea. Thanks!